### PR TITLE
[file_selector_web] migrate to pkg:web

### PR DESCRIPTION
**Pull request by kevmoo**
This allows this package to be used in a web app compiled to Wasm.

Helps unblock https://github.com/flutter/devtools/issues/6606

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/file_selector/file_selector_web/lib/src/dom_helper.dart [hunk 1]
```diff
@@ -3,41 +3,46 @@
...
     bool multiple = false,
-    @visibleForTesting FileUploadInputElement? input,
+    @visibleForTesting HTMLInputElement? input,
   }) {
     final Completer<List<XFile>> completer = Completer<List<XFile>>();
-    final FileUploadInputElement inputElement =
-        input ?? FileUploadInputElement();
+    final HTMLInputElement inputElement =
+        input ?? (createElementTag('input') as HTMLInputElement)
+          ..type = 'file';
```
Comment (ditman) [inline packages/file_selector/file_selector_web/lib/src/dom_helper.dart:33]: It'd be nice if this asserted that `input.type == 'file'`, to retain some of the old type safety.

File: packages/file_selector/file_selector_web/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 0.9.3
+
+* Updates minimum supported SDK version to Dart 3.2.
```
Comment (ditman) [inline packages/file_selector/file_selector_web/CHANGELOG.md:3]:
```suggestion
* Migrates web implementation to `package:web` from `dart:html`.
* Updates minimum supported SDK version to Dart 3.2.
```

**Review by ditman — APPROVED** [review]
LGTM, and CI seems happy, let's go!

File: packages/file_selector/file_selector_web/CHANGELOG.md (hunk 2 above)
Comment (kevmoo) [reply packages/file_selector/file_selector_web/CHANGELOG.md:3]: 🤷 – for the most part, this is not a user visible change so I omitted that bit.

File: packages/file_selector/file_selector_web/lib/src/dom_helper.dart (hunk 1 above)
Comment (kevmoo) [reply packages/file_selector/file_selector_web/lib/src/dom_helper.dart:33]: not worried about this, since it's text only

**Pull request merged**